### PR TITLE
distribute substeps

### PR DESCRIPTION
**Pull request by evykassirer**

### CODE REVIEW

File: lib/distribute.js [hunk 1]
```diff
@@ -100,33 +106,73 @@ function distributeUnaryMinus(node) {
...
+
+    const combinedNode = distributeTwoNodes(
+      clone(firstArg, false), clone(secondArg, false));
+    firstArg.changeGroup = 1;
+    secondArg.changeGroup = 1;
+    combinedNode.changeGroup = 1;
+    let newNode = clone(oldNode, false);
+    // case 1: there were more than two operands in this multiplication
+    // e.g. 3*7*(2+x)*(3+x)*(4+x) would expand just 7*(2+x)
+    // So remove the two args that were combined and replace with the new arg
```
Comment (aelnaiem) [inline lib/distribute.js]: So what would it look like instead?

File: lib/distribute.js [hunk 2]
```diff
@@ -100,33 +106,73 @@ function distributeUnaryMinus(node) {
...
+    }
+
+    const distributionStatus = NodeStatus.nodeChanged(
+      MathChangeTypes.DISTRIBUTE, oldNode, newNode, false);
+
+    if (!simplifyStatus || !simplifyStatus.hasChanged()) {
+      return distributionStatus;
+    }
+    else {
+      const substeps = [distributionStatus, simplifyStatus];
```
Comment (aelnaiem) [inline lib/distribute.js]: subSteps* ? I've been unsure of whether it's one word or two myself

File: lib/distribute.js [hunk 3]
```diff
@@ -176,6 +225,69 @@ function distributeTwoNodes(firstNode, secondNode) {
   return NodeCreator.parenthesis(NodeCreator.operator('+', newArgs));
 }
 
+// Simplifies a sum of terms (a result of distribution) that's in parens
+// (note that all results of distribution are in parens)
+// e.g. 2x*(4 + x) distributes to (2x*4 + 2x*x)
+// This is a separate function from simplify to make the flow more readable,
+// but this is literally just a wrapper around 'simplify'.
+// Returns a NodeStatus object
+function simplifyWithParens(node) {
```
Comment (aelnaiem) [inline lib/distribute.js]: Hmm, I'm not sure that having this as a separate function actually makes this flow more readable

**Comment by aelnaiem** [comment]
👍 a couple comments, otherwise another one bites the dust! I'm struggling to wrap my head around the new structure around this new substep approach. What the pattern and overall code organization is. Would be cool if you wrote it up as part of the blog post

**Comment by evykassirer** [comment]
gonna change the comment before merging

also, will merge equation substeps first and then rename substeps here as well

**Pull request merged**